src/main/java/com/solace/spring/cloud/stream/binder/util/LargeMessageSupport.java

```diff
@@ -0,0 +1,198 @@
+                        log.warn("Check if Queue is partitioned correctly!");
+                        log.warn("Incomplete large message dropped/rejected, no message received within 1 Minute. Dropped chunk {} index {} of {}", o.getKey(), i, o.getValue().length);
+                    }
+                }
+            }
+        } catch (Exception ex) {
+            log.error("Error during housekeeping", ex);
+        }
+    }
+
+    boolean tooOld(MessageContextBytes[] messages) {
+        return Arrays.stream(messages).filter(Objects::nonNull).mapToLong(messageContextBytes -> messageContextBytes.timestamp().toEpochMilli()).max().orElse(0) < Instant.now().toEpochMilli() - RECEIVE_TIMEOUT;
+    }
+
+    public MessageContext assemble(BytesXMLMessage smfMessage, AcknowledgmentCallback acknowledgmentCallback) {
+        try {
+            if (smfMessage.getProperties() == null || !smfMessage.getProperties().containsKey(SolaceBinderHeaders.CHUNK_ID)) {
+                //no large message -> send ahead
+                return new MessageContext(smfMessage, acknowledgmentCallback, Instant.now());
+            }
+            Long chunkId = smfMessage.getProperties().getLong(SolaceBinderHeaders.CHUNK_ID);
+            if (chunkId == null) {
+                throw new RuntimeException("Missing chunkId");
+            }
+            Integer chunkIndex = smfMessage.getProperties().getInteger(SolaceBinderHeaders.CHUNK_INDEX);
+            if (chunkIndex == null) {
+                throw new RuntimeException("Missing chunkIndex");
+            }
+            Integer chunkCount = smfMessage.getProperties().getInteger(SolaceBinderHeaders.CHUNK_COUNT);
+            if (chunkCount == null) {
+                throw new RuntimeException("Missing chunkCount");
+            }
+            if (!(smfMessage instanceof BytesMessage)) {
+                throw new RuntimeException("LargeMessageSupport is only available for BytesMessages");
+            }
+
+            MessageContextBytes[] chunkArray;
+            synchronized (context) {
+                chunkArray = context.computeIfAbsent(chunkId, k -> new MessageContextBytes[chunkCount]);
+            }
+            if (chunkArray[chunkIndex] == null) {
+                chunkArray[chunkIndex] = new MessageContextBytes((BytesMessage) smfMessage, acknowledgmentCallback, Instant.now());
+            } else {
+                log.warn("Duplicate chunk id:{} index:{} of {} received, drop it", chunkId, chunkIndex, chunkCount);
+                return null;
+            }
+            int lenght = 0;
+            for (int i = chunkArray.length - 1; i >= 0; i--) {
+                MessageContextBytes messageContextBytes = chunkArray[i];
+                if (messageContextBytes == null) {
+                    // not all chunks received yet
+                    return null;
+                }
+                lenght += messageContextBytes.bytesMessage().getData().length;

```

Comment:
Limits to 2.7GB Message size due to integer. However should be okey with message size in this use-case.

Reply:
I know but we can't have byte[] bigger anyway so yes, there is a limit.